mono/metadata/threads.c

```diff
@@ -4738,28 +4723,32 @@ mono_thread_execute_interruption (void)
 		ERROR_DECL (error);
 		exc = mono_exception_new_thread_interrupted (error);
 		mono_error_assert_ok (error); // FIXME
+		fexc = TRUE;
 	}
 exit:
 	if (unlock)
 		unlock_thread_handle (thread);
 
-	HANDLE_FUNCTION_RETURN_REF (MonoException, exc);
+	if (fexc && pexc)
+		*pexc = MONO_HANDLE_NEW (MonoException, MONO_HANDLE_RAW (exc));
```

Comment:
This will give the caller a handle that is going to be deallocated by `HANDLE_FUNCTION_RETURN_VAL`, below.

Reply:
Oh um. Can I allocate in the caller's frame?

Reply:
I have an idea...